### ISSUE TITLE
Update Google.Api.CommonProtos generated code

### ISSUE THREAD

**Issue by JamesNK** [googleapis/gax-dotnet#368]
Google.Protobuf has a new API for getting custom options and the types in Google.Api.CommonProtos don't support it because the generated code is from an older version of protoc.

https://github.com/protocolbuffers/protobuf/issues/6956#issuecomment-559212920

**Comment by jskeet** [comment]
Will need to check - some of the more recent Google.Protobuf releases have had issues; if that's been fixed, I'm happy to regenerate.

**Comment by jskeet** [comment]
Right, it looks like the latest GA release is still 3.10.1 which I know has some bugs in (as I fixed them :)
It looks like 3.11.0 should be out soon - at that point I'll update the generator we use (and our dependency).

It will be somewhat awkward to do another release before the start of 2020, however - we're in the middle of planning a large new major version release (prompted by the gRPC major version bump). We definitely *can* do another release, via creating a new branch specifically for it, based on the older version, but I'd *prefer* not to if possible. What's the urgency of this?

**Comment by JamesNK** [comment]
It's not urgent. I'm playing around with a prototype.

The current version of Google.Protobuf seems pretty broken around custom options. I could force the old version, or regenerate `http.proto` and ship it with the library.

**Comment by jskeet** [comment]
From what I remember, you should be able to access the custom options using 3.8.0 (which is what Google.Api.CommonProtos targets), but you need to know the option field number first, and it's awkward. If you'd like me to put together an example of doing it (awkwardly but plausibly) for the current library, let me know.

**Comment by JamesNK** [comment]
It's fine. I'm already doing it with 3.8.0.

```cs
const int HttpRuleFieldId = 72295728;
var success = methodDescriptor.CustomOptions.TryGetMessage<HttpRule>(HttpRuleFieldId, out httpRule);
```

I just want to make sure that there is a solution for new Google.Protobuf versions.

A question I do have is is there a way to support `google.api.http` without requiring people to worry about placing `google/api/annotations.proto` and `google/api/http.proto` at the right location in their project? Basically I want importing those files to Just Work like `import "google/protobuf/empty.proto";` does.

This is for a gRPC projects and the compiler is called by Grpc.Tools and msbuild. I'm guessing I could probably do some cleaver tricks with msbuild to make the protoc compiler happy by automatically supplying the extra proto files during build.

Any ideas or suggestions?

**Comment by jskeet** [comment]
Not sure about the best approach for the second part. I agree it's kinda tricky having to have the right proto source files around. (Deciding between copying them vs git submodules etc.) Although even "import google/protobuf/empty.proto" only works if you've got the core protos in your proto include path... but at least those protos com with the tooling.

One thing we could consider would be to include the source protos in this (Google.Api.CommonProtos) package - that way if you depend on the NuGet package, you get both the generated C# and the source protos to work with. Or if we don't want to conflate the two, we could create a new package (Google.Api.CommonProtos.Source or similar). Thoughts on that?

**Comment by jskeet** [comment]
(Although it's probably worth creating a separate issue for that...)

**Comment by JamesNK** [comment]
Hmm, I forgot about including content in NuGet package as an option. That might work. I'll try including the two file I need in my own library's package.

**Comment by JamesNK** [comment]
@jskeet 3.11.2 is out on NuGet. Are you able to update the generated code now?

**Comment by jskeet** [comment]
It's on my to-do list - but as we've updated targets, I won't want to publish the package just yet. I'll see how hard it would be to revert that target change just to do this...

**Comment by jskeet** [comment]
Done - I've released 2.0.0-beta01 of the package. Please let me know if it doesn't work for you for some reason.
